docs(examples): add colors_rgb example

joshka opened the pull request:
Helpful for diagnosing what's happening with https://github.com/ratatui-org/ratatui/issues/475

![colors_rgb](https://raw.githubusercontent.com/ratatui-org/ratatui/images/examples/colors_rgb.gif)

kdheepak commented on examples/colors.rs:
```diff
@@ -28,32 +27,149 @@ fn main() -> Result<()> {
+
+fn render_title<B: Backend>(frame: &mut Frame<'_, B>, area: Rect) {
+    frame.render_widget(
+        Paragraph::new("Color example. Press Space for RGB, q to quit")
```
Minor comment. It'd be nice to change the title here based on `view_index`.

kdheepak commented:
Do you think this should just be a separate demo?

joshka commented:
> Do you think this should just be a separate demo?

Yes probably. I wasn't sure about it myself, so just went with adding code, but it wouldn't hurt.

orhun commented:
Works great!

It isn't really intuitive from the demo so I'd also say that we should have this as a separate example.

joshka commented:
I tweaked a bunch of stuff in the updated example:
- Modified color grid to be smooth by mirroring blue channel every 16 pixels instead of starting at 0
- Changed to `buf.cell_mut()` this is noticeably faster to render than using `Paragraph`
- App-ified and Widget-ified implementation
  - Removed manual Error printing (returning `Result<()>` handles this automatically)
  - panic hook and `Drop` impl (because I kept getting index out of bounds errors when playing around with generating the grid)
- Simplified everything as much as possible

![Made with VHS](https://vhs.charm.sh/vhs-2KH9mVYokPvHplFYZ5tHoM.gif)

joshka commented:
- Added example to cargo toml and fixed commit message

orhun commented on examples/colors_rgb.rs:
```diff
@@ -0,0 +1,159 @@
+type Result<T> = std::result::Result<T, Box<dyn Error>>;
+
+fn main() -> Result<()> {
+    better_panic::install();
```
Wow, nice crate! Should we also use this for the other examples for backtracing errors etc.?

kdheepak replied:
I think `better_panic` (or `human_panic`) should be part of all our examples + templates etc. It is a much nice developer experience and user experience with these installed.

I do have [a section in the book](https://ratatui-org.github.io/ratatui-book/how-to/setup-panic-hooks-better-panic.html) that describes how users can use `better_panic` so that's a start.

joshka replied:
Yep definitely - I'm searching for a consistent approach to use for all the examples instead of something different in each one. I think this is pretty close to the right balance of simplicity and functionality for examples. It's just enough to not have to pass around too much state. 

I wonder though whether the `init_terminal()`, `restore_terminal()` and `install_panic_hook()` methods might be worth putting at the bottom of the code outside of `impl App` instead.

joshka commented:
moved the init/restore/panic hook to the bottom of the file to make the example really hyper focus on the application logic and example rather than the generic terminal parts.
